apps/server/src/modules/provisioning/service/provisioning.service.ts

```diff
@@ -0,0 +1,55 @@
+import { Injectable, InternalServerErrorException, UnprocessableEntityException } from '@nestjs/common';
+import { SystemProvisioningStrategy } from '@shared/domain/interface/system-provisioning.strategy';
+import { Logger } from '@src/core/logger';
+import { ProvisioningSystemInputDto } from '@src/modules/provisioning/dto/provisioning-system-input.dto';
+import { ProvisioningSystemInputMapper } from '@src/modules/provisioning/mapper/provisioning-system-input.mapper';
+import { SanisProvisioningStrategy, SanisStrategyData } from '@src/modules/provisioning/strategy/sanis/sanis.strategy';
+import { IservProvisioningStrategy, IservStrategyData } from '@src/modules/provisioning/strategy/iserv/iserv.strategy';
+import { ProvisioningDto } from '@src/modules/provisioning/dto/provisioning.dto';
+import { SystemService } from '@src/modules/system/service/system.service';
+
+@Injectable()
+export class ProvisioningService {
+	constructor(
+		private readonly systemService: SystemService,
+		private readonly sanisStrategy: SanisProvisioningStrategy,
+		private readonly iservStrategy: IservProvisioningStrategy,
+		private readonly logger: Logger
+	) {
+		this.logger.setContext(ProvisioningService.name);
+	}
+
+	async process(accessToken: string, idToken: string, systemId: string): Promise<ProvisioningDto> {
+		let system: ProvisioningSystemInputDto;
+		try {
+			system = ProvisioningSystemInputMapper.mapToInternal(await this.systemService.findById(systemId));
+		} catch (e) {
+			this.logger.error(`System with id ${systemId} was not found.`);
+			throw new UnprocessableEntityException(`System with id "${systemId}" does not exist.`);
+		}
+
+		switch (system.provisioningStrategy) {
+			case SystemProvisioningStrategy.SANIS: {
+				if (!system.provisioningUrl) {
+					throw new UnprocessableEntityException(`Sanis system with id: ${systemId} is missing a provisioning url`);
```

Comment:
Sounds like one of the 500 errors and a misstake of our side.

Reply:
Its now an InternalServerErrorException